src/adapters/supabase/helpers/wallet.ts

```diff
@@ -42,21 +41,35 @@ export class Wallet extends Super {
     }
   }
 
+  public async unlinkWalletFromUserId(userId: number) {
+    const userData = await this._getUserWithWallet(userId);
+
+    if (!userData.wallet_id) {
+      throw this.context.logger.error("The user does not have an associated wallet to unlink");
+    }
+
+    const { error } = await this.supabase.from("users").update({ wallet_id: null }).eq("id", userData.id);
```

Comment:
Did you test that this works? My original belief was that we would clobber their existing wallet but this approach seems more correct especially for audits- being able to save every wallet. But as an extension of that logic it might make sense to have a record of all the IDs historically for any user too?

Reply:
> Did you test that this works? My original belief was that we would clobber their existing wallet but this approach seems more correct especially for audits- being able to save every wallet. But as an extension of that logic it might make sense to have a record of all the IDs historically for any user too?

All wallets get saved but once the user changes it or unsets it, we lose the connection between wallet and user so we could add a userId to the wallet along with start and end date so we can have a history of all wallets and when they were used

> One other edge case problem to consider but when I demo this capability I set mine to ubq.eth then back to mine. If I do this, then I'll have a lot of redundant entries.

There won't be redundant entries because it searches for existing wallet and just updates `wallet_id` on the user with existing wallet `id`. However if we implement it like I said above that would cause many entries if you're testing

Reply:
Can think of this in a separate PR. We can have a new table that would keep history of permit associations timelimes, to avoid duplicating permit entries.